tutorials/3d/portals/fundamentals.rst

```diff
@@ -0,0 +1,102 @@
+
+.. image:: img/room_manager.png
+
+Rooms
+~~~~~
+What is a room?
+^^^^^^^^^^^^^^^
+:ref:`Room<class_Room>`\ s are a way of spatially partitioning your level into areas that make sense in terms of the level design. Rooms often quite literally *are* rooms (for instance in a building). Ultimately though, as far as the engine is concerned, a room respresents a **non-overlapping** convex volume, in which you would typically place most of your objects that fall within that area.
+
+A room doesn't need to correspond to a literal room. It could for example also be a canyon in an outdoor area, or a smaller part of a concave room. With a little imagination you can use the system in almost any scenario.
+
+Why convex?
+^^^^^^^^^^^
+The reason why rooms are defined as convex volumes (or 'convex hulls' as they are known), is that mathematically it is very easy to determine whether a point is within a convex hull. A simple plane check will tell you the distance of a point from a plane. If a point is behind all the planes bounding the convex hull, then by definition, it is inside the room. This makes all kinds of things easier in the internals of the system, such as checking which room a camera is within.
+
+*A convex hull. The hull is defined as a series of planes facing outward. If a point is behind all the planes, it is within the hull.*
+
+.. image:: img/convex_hull.png
+
+Why non-overlapping?
+^^^^^^^^^^^^^^^^^^^^
+If two rooms overlap, and a camera or player is in this overlapping zone, then there is no way to tell which room the object should be in - and hence render from, or be rendered in. This requirement for non-overlapping rooms does have implications for level design.
+
+If you accidentally create overlapping rooms, the editor will flag a warning when you convert the rooms, and indicate any overlapping zones in red.
+
+.. image:: img/room_overlap.png
+
+The system does attempt to cope with overlapping rooms as best as possible by making the current room *'sticky'*. That is, each object remembers which room it was in last frame, and stays within it as long as it does not move outside the convex hull room bound. This can result in some hysteresis in these overlapping zones.
+
+There is one exception however - *internal rooms* (*they are described later, you do not have to worry about these to start with*).
+
+How do I create a room?
+^^^^^^^^^^^^^^^^^^^^^^^
+A :ref:`Room<class_Room>` is a node type that can be added to the scene tree like any other. You can place objects within the room by making them children and grand-children of the Room node. Instead of placing the rooms as children of a scene root node, you will need to create a Spatial especially for the job of being the parent of all the rooms. This node we will refer to as the ``RoomList``. You will need to assign the roomlist node in the RoomManager, so the RoomManager knows where to find the rooms.
```

Comment:
```suggestion
How do I create a room?
^^^^^^^^^^^^^^^^^^^^^^^

A :ref:`Room<class_Room>` is a node type that can be added to the scene tree like any other. You can place objects within the room by making them children and grand-children of the Room node. Instead of placing the rooms as children of a scene root node, you will need to create a Spatial especially for the job of being the parent of all the rooms. This node we will refer to as the ``RoomList``. You will need to assign the roomlist node in the RoomManager, so the RoomManager knows where to find the rooms.
```